src/Services/PatientService.php

```diff
@@ -44,10 +47,37 @@ public function validate($patient)
         $validator->required('sex')->lengthBetween(4, 30);
         $validator->required('DOB')->datetime('Y-m-d');
 
+        return $validator->validate($patient);
+    }
+
+    public function validateUpdate($pid, $patient)
+    {
+        $validator = new Validator();
+
+        $vPid = $this->validatePid($pid);
+        if ($vPid->isNotValid()) {
+            return $vPid;
+        }
+
+        $validator->optional('fname')->lengthBetween(2, 255);
+        $validator->optional('lname')->lengthBetween(2, 255);
+        $validator->optional('sex')->lengthBetween(4, 30);
+        if (key_exists('DOB', $patient)) {
+            $validator->required('DOB')->datetime('Y-m-d');
+        }
```

Comment:
Oops I missed this one.
Isn't optional works only when the key_exisits?
http://validator.particle-php.com/en/latest/required-optional/

Reply:
If DOB is passed in, we want to require verification otherwise ignore. Update default should not require DOB or anything accept pid and maybe row id. This is what started us down this road, remember:)

Reply:
Yeah But if we use this
`$validator->optional('DOB')->datetime('Y-m-d');` 
this will also do the same (check if DOB is passed then validate otherwise ignore)
![image](https://user-images.githubusercontent.com/47683556/78597935-fece7b00-786b-11ea-9a8b-229820c34327.png)
If user is not passing any value validator is returnin TRUE.

Reply:
if an empty DOB or not properly formatted value is pass to validation as optional if not specifically checked in the response, it will pass through with how we check in controller now.
Try it by doing a put with dob set to "" and validate set to optional. I'm sure I tested this

Reply:
Wired! Empty is getting passed but single space or anything unformatted doesn't work 😕

Reply:
Worrisome huh! Could be a bug but you'll figure it out

Reply:
Yeah! allowedEmpty is by default `true` IDK why :<
As per guide, it should be `false` default.

Reply:
$validator->optional('DOB', 'DOB', false)->datetime('Y-m-d');
😆 This works fine in every case.